FAQ.md

```diff
@@ -0,0 +1,119 @@
+```
+
+For comparison, in Agda the definition would be:
+
+```agda
+data Vec (A : Set) : (len : Nat) -> Set where
+  nil : Vec A zero
+  cons : (n : Nat) -> (x : A) -> (xs : Vec A n) -> Vec A (succ n)
+```
+A few differences: in Agda, you separate parameters and indices with a : . In Formality, you write parameters inside <> and indices after the ~ . In Agda, you need -> Set on the first line. In Formality, no. In Agda, you write the fields with (a : A) -> (b : B) -> ... . In Formality, you write with (a: A, b: B) . In Agda, you need to write the return type of each constructor. In Formality, we omit the return type. Problem is, that only works for non-indexed datatypes; for indexed datatypes, you *need* to write the indices. So, to do that while also omitting the return type, we use ~ to configure the index on each case.
+
+### Does Formality compile to native datatypes by name or definition?
+
+The Javascript backend for Formality automatically compiles `Bool`'s
+`String`'s, `Nat`'s, etc. to native Javascript booleans, strings, BigIntegers, etc.
+This produces large performance gains over lambda-encoded data structures.
+Formality uses the name of the type to do this - i.e, `Nat` will always be compiled
+to BigIntegers, but an identical definition with a different name won't.
+
+### How do I implement interfaces/typeclasses/traits?
+
+A good example to emulate is [Moonad's definition for monad](https://github.com/moonad/Moonad/blob/master/lib/Monad.fm). The basic idea is to define a seperate type that
+takes in a type as an erased argument, and whose type constructor takes in the methods
+defined on that "typeclass/trait". Then you can define "methods" on the "typeclass"
+that just call the implementation supplied by the user.
+
+(note: this will change if and when Formality has functional extensionality)
+
+### Is Formality Consistent?
+
+No, and that's by design. Here's a snippet from a recent conversation on it:
+
+> ...I concluded making core consistent was a software engineering mistake, since it forces everyone into a specific termination discipline, and different people might want to work on different terminating subsets. For example you might have structural recursion and be Coq/Agda-like. Or you might have elementary duplications and be interaction-net friendly. Or you may just be writing web apps and not care about termination.
+
+### Can I have axioms?
+
+Definitely! It's as easy as recursion:
+
+```
+axiom: A
+    axiom
+```
+
+This obviously wouldn't pass the termination checker, is still valid Formality code.
+
+### What are the different parts of Formality?
+- Formality-Lang
+- Formality-Core
+- The termination checker (to be implemented)
+- A backend (Javascript, Haskell, EVM, etc.)
+- Moonad
+
+### What's the difference between Formality-Core and EA-TT?
+- FormalityCore is a superset of EA-TT.
+- FormalityCore with boxes is EA-TT. This subset has affinity and termination, and compiles to the super-fast inets.
```

Comment:
These two statements seem to contradict each other. The first says FormalityCore = EA-TT + something, while the second says FormalityCore + boxes = EA-TT.

Reply:
We might need @MaiaVictor to clarify, but I think "FormalityCore with boxes" is interpreted as "FormalityCore restricted to that subset compatible with boxes". I mostly included this (and all the rest) so we could get a good a sense of the kinds of questions that are coming up in the chat - all the answers need more detail/work.

Reply:
There's no contradiction, boxes are a restriction on core, not an addition

Reply:
"with" implies addition. Maybe write "FormalityCore restricted to boxes"?

Reply:
Okay, so we have to distinguish between a syntactic addition and a semantic addition. A box is just a way to wrap a value in some extra information. In EA-TT, where all variables are affine (used 0 or 1 times) we use boxes to indicate duplication:

Formality-Core:
````
Demo.duplicate<A: Type>(x: A) : Pair(A,A)
   Pair.new<A,A>(x,x)
```

EA-TT:

```
duplicate<A: Type>(x: !A) : Pair(!A,!A)
   dup x = x
   Pair.new<!A,!A>(#x,#x)
```

EA-TT syntactically adds the `!` boxed type and `#` boxed value syntax, but this restricts the semantic expressivity of Core, since now we need different `duplicate` functions for the above and:

```
duplicate<A: Type>(x: !A) : !Pair(A,A)
   dup x = x
   #Pair.new<A,A>(x,x)
```